network/endpoint_windows.go

```diff
@@ -18,6 +18,8 @@ import (
 	"github.com/Microsoft/hcsshim/hcn"
 )
 
+var hnsv2 hnswrapper.HnsV2WrapperInterface = hnswrapper.Hnsv2wrapper{}
```

Comment:
Instead of initializing it here, can you pass this in for each needed call. And let this be part of NetworkManager. We can initialize this in NewnetworkManager. That way we can control what gets passed on early on.

Reply:
I want to avoid passing around os specific code such as HNS in platform agnostic code. By passing from the top down, its my opinion that it muddies the code. Happy to chat more about it though.